crates/router/src/connector/cryptopay/transformers.rs

```diff
@@ -0,0 +1,173 @@
+use reqwest::Url;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    connector::utils::CryptoData,
+    core::errors,
+    services,
+    types::{self, api, storage::enums},
+};
+
+#[derive(Default, Debug, Serialize)]
+pub struct CryptopayPaymentsRequest {
+    price_amount: i64,
+    price_currency: enums::Currency,
+    pay_currency: String,
+    success_redirect_url: Option<String>,
+    unsuccess_redirect_url: Option<String>,
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for CryptopayPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        let cryptopay_request = match item.request.payment_method_data {
+            api::PaymentMethodData::Crypto(ref cryptodata) => {
+                let pay_currency = cryptodata.get_pay_currency()?;
+                Ok(Self {
+                    price_amount: item.request.amount,
+                    price_currency: item.request.currency,
+                    pay_currency,
+                    success_redirect_url: item.clone().request.router_return_url,
+                    unsuccess_redirect_url: item.clone().request.router_return_url,
+                })
+            }
+            _ => Err(errors::ConnectorError::NotImplemented(
+                "payment method".to_string(),
+            )),
+        }?;
+        Ok(cryptopay_request)
+    }
+}
+
+// Auth Struct
+pub struct CryptopayAuthType {
+    pub(super) api_key: String,
+    pub(super) api_secret: String,
```

Comment:
Wrap these values within `Secret`.

Reply:
Hey, we are not wraping any of the connecter's auth keys into secret.